filters/setupresponse.go

```diff
@@ -0,0 +1,73 @@
+package filters
+
+import (
+	"bytes"
+	log "github.com/Sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	"gopkg.in/h2non/bimg.v1"
+	"io/ioutil"
+	"net/http"
+)
+
+// This filter is the default filter for every configuration of skrop
+// It
+type setupResponse struct{}
+
+const (
+	SetupResponseName = "setupResponse"
+	SkropImage        = "skImage"
+	SkropOptions      = "skOptions"
+)
+
+func NewSetupResponse() filters.Spec {
+	return &setupResponse{}
+}
+
+func (s *setupResponse) Name() string {
+	return SetupResponseName
+}
+
+func (s *setupResponse) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) != 0 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	return &setupResponse{}, nil
+}
+
+func (s *setupResponse) Request(ctx filters.FilterContext) {}
+
+//initialize the statebag where the options created by each filter and the image are stored
+func (s *setupResponse) Response(ctx filters.FilterContext) {
+	rsp := ctx.Response()
+
+	defer rsp.Body.Close()
+
+	rsp.Header.Del("Content-Length")
+
+	buf, err := ioutil.ReadAll(rsp.Body)
+	imageBytesLength := len(buf)
+
+	log.Debug("Image bytes length: ", imageBytesLength)
+
+	if err != nil {
+		log.Error("Failed to process image ", err.Error())
+		ctx.Serve(&http.Response{
+			StatusCode: http.StatusInternalServerError,
+			Body:       ioutil.NopCloser(bytes.NewBufferString(err.Error())),
```

Comment:
i think it can be dangerous to dump error messages to the public because it can give hints for attackers. It is enough if we log it and use a default message for 500.